gempython/tools/amc_user_functions_xhal.py

```diff
@@ -524,49 +564,62 @@ def getVFATLinkStatus(self,doReset=False,printSummary=False, ohMask=None):
         Get's the VFAT link status and can print a table of the status for each unmasked OH.
         Returns True if all unmasked OH's have all VFAT's with:
             1. SYNC_ERR_CNT = 0x0
-        
+
         doReset - Issues a link reset if True
         printSummary - prints a table summarizing the status of the GBT's for each unmasked OH
         ohMask - Mask which defines which OH's to query; 12 bit number where
                  having a 1 in the N^th bit means to query the N^th optohybrid.
                  If None will be determined automatically using HwAMC::getOHMask()
         """
-        
+
         # Automatically determine ohMask if not provided
         if ohMask is None:
             ohMask = self.getOHMask(callingMthd="getVFATLinkStatus")
 
         vfatMonData = VFATLinkMonitorArrayType()
         self.getmonVFATLink(vfatMonData, self.nOHs, ohMask, doReset)
 
-        if(printSummary):
+        if printSummary:
             print("--=======================================--")
             print("-> GEM SYSTEM VFAT INFORMATION")
             print("--=======================================--")
             print("")
             pass
+            hfmt      = "{:4s}"
+            vfatsyncfmt = ["VFAT{}.SYNC_ERR_CNT".format(vfat) for vfat in range(24) ]
+
+            stfmt = "{}0x{:1x}{}"
+            lines = [[] for x in range(len(vfatsyncfmt)+1)]
+            lines[0].append("{}".format(" "*(len(max(vfatsyncfmt,key=len)))))
+            for st in range(len(vfatsyncfmt)):
+                lines[st+1].append("{{:{}s}}".format(len(max(vfatsyncfmt,key=len))).format(vfatsyncfmt[st]))
 
         totalSyncErrors = 0
         for ohN in range(self.nOHs):
             # Skip Masked OH's
             if (not ((ohMask >> ohN) & 0x1)):
                 continue
 
-            if(printSummary):
-                print("----------OH{0}----------".format(ohN))
+            if printSummary:
+                lines[0].append(hfmt.format(("OH{:d}".format(ohN)).rjust(4)))
+                # print("----------OH{0}----------".format(ohN))
                 pass
 
             for vfatN in range(24):
                 nSyncErrors = vfatMonData[ohN].syncErrCnt[vfatN]
                 totalSyncErrors += nSyncErrors
 
-                if(printSummary):
-                    print("VFAT{0}.SYNC_ERR_CNT {1}{2}{3}".format(
-                        vfatN,colors.RED if (nSyncErrors > 0) else colors.GREEN,hex(nSyncErrors),colors.ENDC))
+                if printSummary:
+                    lines[vfatN+1].append(hfmt.format(stfmt.format(colors.RED if (nSyncErrors > 0) else colors.GREEN, nSyncErrors, colors.ENDC)))
+                    # print("VFAT{0}.SYNC_ERR_CNT {1}{2}{3}".format(
+                    #     vfatN,colors.RED if (nSyncErrors > 0) else colors.GREEN,hex(nSyncErrors),colors.ENDC))
```

Comment:
comment lines were intended to be committed?

Reply:
Were unremoved, as they were part of the old style i used to get the syntax right